pkgs/development/libraries/cpp-utilities/default.nix

```diff
@@ -18,19 +19,23 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
   nativeCheckInputs = [ cppunit ];
+  buildInputs = lib.optionals stdenv.isDarwin [
+    iconv # needed on Darwin, see https://github.com/Martchus/cpp-utilities/issues/4
+  ];
   # Otherwise, tests fail since the resulting shared object libc++utilities.so is only available in PWD of the make files
   preCheck = ''
     checkFlagsArray+=(
       "LD_LIBRARY_PATH=$PWD"
     )
   '';
-  doCheck = true;
+  # tests fail on Darwin
+  doCheck = stdenv.isLinux;
```

Comment:
How about:

```suggestion
  doCheck = !stdenv.isDarwin;
```

Reply:
I personally don't believe it should be `!stdenv.isDarwin` because the tests can fail on other platforms added in the future as well (e.g. `mingwW64`)

Reply:
> I personally don't believe it should be `!stdenv.isDarwin` because the tests can fail on other platforms added in the future as well (e.g. `mingwW64`)

Yes but ideally upstream could improve the situation and make the checks succeed in the future. This is how we force our future selves to check the tests when we add another platform.

Reply:
BTW if you have more information / ideas about why these tests fail on Darwin, it'd be nice to add them in a comment, but it's not mandatory.